src/definitions.jl

```diff
@@ -0,0 +1,74 @@
+
+using LightGraphs, MetaGraphs
+
+import JSON
+
+# TODO: Create functions for each
+# Cause - is concept label, Effect - definition label
+# within a RelationMention with labels "Definition, Entity"
+
+definitions_output_file = "/Users/scott/Documents/JuliaProjects/SemanticModels.jl/test/definitions.jl/automates_output_1sentence_definition.json"
+def_output_dir = "/Users/scott/Documents/JuliaProjects/SemanticModels.jl/test/definitions.jl"
+
+
+function info_extract_from_automates(output_dir::String)
+    file_list = readdir(output_dir)
+    graph = DiGraph()
+    changes_recorded = []
+    v_count = 0
+    for def_file in file_list
+        json_res = JSON.parsefile(string(def_output_dir, "/",def_file))
+        # json_res = JSON.parsefile(definitions_output_file)
+        mentions = json_res["mentions"]
+        # println("JSON Mentions is:\n")
+        # println(mentions)
+        # println("End mentions\n")
+        for val in mentions
+            if haskey(val,"type") && val["type"] == "RelationMention"
+                JSON.print(val, 4)
+                v_count += 1
+                cause = val["arguments"]["cause"][1]["text"]
+                effect = val["arguments"]["effect"][1]["text"]
+                println(string("Cause is: ", cause, "\n"))
+                println(string("Effect is: ", effect, "\n"))
+                println("Add new vertex for Concept to Graph")
+                add_vertex!(graph)
+                push!(changes_recorded, Dict(:v_id=>v_count, :type=> "AddVertex", :values => Dict(:name=> string(cause, hash(effect)), :id => v_count, :cause => cause)))
+                v_count += 1
+                add_vertex!(graph)
+                push!(changes_recorded, Dict(:v_id=>v_count, :type=> "AddVertex", :values => Dict(:name=> string("Def_",hash(effect)), :id => v_count, :definition => effect)))
+                println("Adding edge from Concept to Definition")
+                add_edge!(graph, v_count-1, v_count)
+                push!(changes_recorded, Dict(:edge=> Edge(v_count-1,v_count), :type=> "AddEdge", :values=> Dict(:name=> "is defined by")))
```

Comment:
Instead of using a Dict with symbol keys, you can use a NamedTuple which is faster for small collections.

`(edge=Edge(v_count-1,v_count), type="AddEdge", values=(name="is defined by"))`

Then you access the named tuple with `tuple.field` notation like a struct or object.
Since your tuples have different structure based on the `type`, you should probably separate them into two lists `vertex_changes` and `edge_changes`.